CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -1,14 +1,23 @@
 #ifndef CUDADataFormats_Track_TrackHeterogeneousT_H
 #define CUDADataFormats_Track_TrackHeterogeneousT_H
 
+#include <string>
+#include <algorithm>
+
 #include "CUDADataFormats/Track/interface/TrajectoryStateSoAT.h"
 #include "HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h"
 
 #include "CUDADataFormats/Common/interface/HeterogeneousSoA.h"
 
 namespace pixelTrack {
-  enum class Quality : uint8_t { bad = 0, dup, loose, strict, tight, highPurity };
-}
+  constexpr uint32_t qualitySize{7};
+  enum class Quality : uint8_t { bad = 0, edup, dup, loose, strict, tight, highPurity, notQuality = qualitySize };
```

Comment:
this could be made directly
```suggestion
  enum class Quality : uint8_t { bad = 0, edup, dup, loose, strict, tight, highPurity, qualitySize };
```
so that `qualitySize` does not need to be kept automatically in sync.

Or, if a separate variable is needed, at least
```suggestion
  enum class Quality : uint8_t { bad = 0, edup, dup, loose, strict, tight, highPurity, notQuality };
  constexpr uint32_t qualitySize = Quality::notQuality;
```
that this requires that `notQuality` is always the last one.

Reply:
I was in doubt and decided to keep it separate.
I think I have a static_assert somewhere